shell/platform/android/io/flutter/embedding/engine/FlutterEngine.java

```diff
@@ -344,6 +361,28 @@ private boolean isAttachedToJni() {
     return flutterJNI.isAttached();
   }
 
+  /**
+   * Create a second {@link FlutterEngine} based on this current one by sharing as much resources
+   * together as possible to minimize startup latency and memory cost.
+   *
+   * @param dartEntrypoint specifies the {@link DartEntrypoint} the new engine should run. It
+   *     doesn't need to be the same entrypoint as the current engine but must be built in the same
+   *     AOT or snapshot.
+   * @return a new {@link FlutterEngine}.
+   */
+  @NonNull
+  /*package*/ FlutterEngine spawn(@NonNull DartEntrypoint dartEntrypoint) {
+    if (!isAttachedToJni()) {
+      throw new IllegalStateException(
+          "Spawn can only be called on a fully constructed FlutterEngine");
+    }
+
+    FlutterJNI newFlutterJNI =
+        flutterJNI.spawn(
+            dartEntrypoint.dartEntrypointFunctionName, dartEntrypoint.dartEntrypointLibrary);
+    return new FlutterEngine(context, null, newFlutterJNI);
```

Comment:
nit: pull `null` into a local variable to document what it is.  It doesn't look like they use parameter comments in java =T

Reply:
It's an IDE feature. Added comments.